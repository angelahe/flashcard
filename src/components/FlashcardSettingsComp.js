import React from 'react';
import '../styles/flashcard.css';
import ActionBarComp from './ActionBarComp';

class FlashcardSettingsComp extends React.Component {
  constructor() {
    super();
    this.state = {
      headerText: 'Settings',
    };
  }

  render() {
    const { headerText } = this.state;
    return (
      <div className="FlashcardSettingsComp">
        <ActionBarComp
          showAdd={false}
          showBack={false}
          headerText={headerText}
        />
        <p>FlashcardSettings coming soon</p>
      </div>
    );
  }
}
export default FlashcardSettingsComp;
