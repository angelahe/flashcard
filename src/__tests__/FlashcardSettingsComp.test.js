import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import FlashcardSettingsComp from '../components/FlashcardSettingsComp';

it('renders without crashing', () => {
  shallow(<FlashcardSettingsComp />);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<FlashcardSettingsComp />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('snapshot of UI renders consistently', () => {
  const tree = renderer.create(<FlashcardSettingsComp />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains the container div FlashcardSettingsComp on shallow render', () => {
  const component = shallow(<FlashcardSettingsComp />);
  expect(component.html()).toContain('FlashcardSettingsComp');
});
