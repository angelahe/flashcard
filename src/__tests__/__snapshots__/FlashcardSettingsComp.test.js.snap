// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of UI renders consistently 1`] = `
<div
  className="FlashcardSettingsComp"
>
  <div
    className="ActionBar ActionBarComp"
  >
    <img
      alt="logo"
      className="AppLogo NavImg NavBtn"
      src="logo_FFFFFF.png"
    />
    <span
      className="HeaderText"
    >
      Settings
    </span>
  </div>
  <p>
    FlashcardSettings coming soon
  </p>
</div>
`;
